### PR TITLE
Treat high priority but syncing peers as low priority

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
This change treats peers as if they have a priority of 0 while they're syncing, for the purpose of deciding who should be mastering.

Fixes: https://github.com/Expensify/Expensify/issues/86480

The idea is to prevent the whole cluster from waiting for the highest priority peer to stand up when it's significantly behind.

Unfortunately, there are no new tests for this. There's not an easy way to test it.

### CODE REVIEW

**Comment by quinthar** [comment]
I vote we save this one for post SB

File: sqlitecluster/SQLiteNode.cpp [hunk 1]
```diff
@@ -520,11 +521,14 @@ bool SQLiteNode::update() {
                         freshestPeer = peer;
 
                     // See if it's the highest priority
-                    if (!highestPriorityPeer || peer->calc("Priority") > highestPriorityPeer->calc("Priority"))
+                    const string& peerState = (*peer)["State"];
+                    int peerEffectivePriority = SIEquals(peerState, "SYNCHRONIZING") ? 0 : peer->calc("Priority");
```
Comment (coleaeason) [inline sqlitecluster/SQLiteNode.cpp]: i think this needs to be 1 instead of 0 or it will cause some oddness with permaslave flows.

Comment (tylerkaraszewski) [reply sqlitecluster/SQLiteNode.cpp]: What oddness?

**Comment by tylerkaraszewski** [comment]
No longer held for SB.

**Comment by tylerkaraszewski** [comment]
@coleaeason - bump?

File: sqlitecluster/SQLiteNode.cpp (hunk 1 above)
Comment (coleaeason) [reply sqlitecluster/SQLiteNode.cpp]: I'm not sure I can point to any -- just that there are places where 0 is used to denote what should be a permaslave and what should not be.

**Pull request merged**